executor/runtime/docker/docker.go

```diff
@@ -1593,12 +1623,13 @@ func (r *DockerRuntime) Start(parentCtx context.Context, pod *v1.Pod) (string, *
 		details.NetworkConfiguration.TransitionIPAddress = a.Address.Address
 	}
 
-	mainTiniConn, err := r.waitForTiniConnection(ctx, listeners[runtimeTypes.MainContainerName])
+	tiniConns, err := r.waitForTiniConnections(ctx, listeners)
 	if err != nil {
 		eventCancel()
 		err = fmt.Errorf("container prestart error: %w", err)
 		return "", nil, statusMessageChan, err
 	}
+	mainTiniConn := tiniConns[runtimeTypes.MainContainerName]
```

Comment:
This will be nil in the Docker for Mac case. Is that okay?

Reply:
Not that I use Linux, eh? 
I think was probably the only one running the standalone tests locally on my laptop (I originally made the docker-for-mac compatibility things).

But yes, if nil the other functions should just fail early. I think I've already sprinkled those in.